com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/Raytracing/HDDiffuseDenoiser.cs

```diff
@@ -144,5 +146,46 @@ static public void DenoiseBuffer(CommandBuffer cmd, DiffuseDenoiserParameters dd
                 cmd.DispatchCompute(ddParams.diffuseDenoiserCS, ddParams.gatherKernel, numTilesX, numTilesY, ddParams.viewCount);
             }
         }
+
+        class DiffuseDenoiserPassData
+        {
+            public DiffuseDenoiserParameters parameters;
+            public TextureHandle depthStencilBuffer;
+            public TextureHandle normalBuffer;
+            public TextureHandle noisyBuffer;
+            public TextureHandle intermediateBuffer;
+            public TextureHandle outputBuffer;
+        }
+
+        public TextureHandle Denoise(RenderGraph renderGraph, HDCamera hdCamera, DiffuseDenoiserParameters tfParameters, TextureHandle noisyBuffer, TextureHandle depthPyramid, TextureHandle normalBuffer, TextureHandle outputBuffer)
+        {
+            using (var builder = renderGraph.AddRenderPass<DiffuseDenoiserPassData>("DiffuseDenoiser", out var passData, ProfilingSampler.Get(HDProfileId.DiffuseFilter)))
+            {
+                // Cannot run in async
+                builder.EnableAsyncCompute(false);
+
+                // Fetch all the resources
+                passData.parameters = tfParameters;
+                passData.depthStencilBuffer = builder.ReadTexture(depthPyramid);
+                passData.normalBuffer = builder.ReadTexture(normalBuffer);
+                passData.noisyBuffer = builder.ReadTexture(noisyBuffer);
+                passData.intermediateBuffer = builder.ReadTexture(builder.WriteTexture(renderGraph.CreateTexture(new TextureDesc(Vector2.one, true, true)
```

Comment:
intermediateBuffer seems to be only used in this pass so in this case you can use CreateTransientTexture and remove Read/WriteTexture. Transient resources are only available for this pass and have implicit read/write operations.